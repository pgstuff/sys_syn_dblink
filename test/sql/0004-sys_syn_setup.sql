CREATE EXTENSION sys_syn;

CREATE SCHEMA user_data
    AUTHORIZATION postgres;

CREATE TABLE user_data.test_table (
        test_table_id integer NOT NULL,
        test_table_text text,
        CONSTRAINT test_table_pid PRIMARY KEY (test_table_id));

INSERT INTO sys_syn.in_groups_def VALUES ('in');

DO $$BEGIN
        EXECUTE sys_syn.in_table_add_sql('user_data.test_table'::regclass, 'in');
END$$;

INSERT INTO user_data.test_table(
        test_table_id, test_table_text)
VALUES  (1,              'test_data 1'),
        (2,              'test_data 2'),
        (3,              'test_data 3');

INSERT INTO sys_syn.out_groups_def VALUES ('out');

SELECT sys_syn.out_table_add('user_data', 'test_table', 'out', data_view => TRUE);

SELECT user_data.test_table_pull(FALSE);
SELECT user_data.test_table_out_move();

SELECT id, delta_type, queue_state FROM user_data.test_table_out_queue;

CREATE EXTENSION dblink;



CREATE TABLE public.test_data (
        test_data_id integer NOT NULL,
        test_data_text text,
        CONSTRAINT test_data_pid PRIMARY KEY (test_data_id));

INSERT INTO sys_syn.in_groups_def VALUES ('in_group');

DO $$BEGIN
        EXECUTE sys_syn.in_table_add_sql('public.test_data'::regclass, 'in_group');
END$$;

INSERT INTO public.test_data(
        test_data_id, test_data_text)
VALUES  (1,              'test_data 1'),
        (2,              'test_data 2'),
        (3,              'test_data 3');

INSERT INTO sys_syn.out_groups_def VALUES ('group');

SELECT sys_syn.out_table_add('public', 'test_data', 'group', data_view => TRUE);

SELECT public.test_data_pull(FALSE);
SELECT public.test_data_group_move();

SELECT id, delta_type, queue_state FROM public.test_data_group_queue;



CREATE SCHEMA "User Data"
    AUTHORIZATION postgres;

CREATE TABLE "User Data"."Test Table" (
        "Test Table ID" integer NOT NULL,
        "Test Table Text" text,
        CONSTRAINT "Test Table_pid" PRIMARY KEY ("Test Table ID"));

INSERT INTO sys_syn.in_groups_def VALUES ('In Group');

SELECT sys_syn.in_table_add_sql('"User Data"."Test Table"'::regclass, 'In Group');

DO $$BEGIN
        EXECUTE sys_syn.in_table_add_sql('"User Data"."Test Table"'::regclass, 'In Group');
END$$;

INSERT INTO "User Data"."Test Table"(
        "Test Table ID", "Test Table Text")
VALUES (1,              'test_data v1');

INSERT INTO sys_syn.out_groups_def VALUES ('Out Group');

DO $$BEGIN
        EXECUTE sys_syn.out_table_add_sql('"User Data"'::regnamespace, 'Test Table', 'Out Group', data_view => TRUE);
END$$;

SELECT "User Data"."Test Table_pull"(FALSE);
SELECT "User Data"."Test Table_Out Group_move"();



CREATE TABLE user_data.test_table_array (
        test_table_array_id integer NOT NULL,
        test_table_array_updated timestamp with time zone,
        test_table_array_text text,
        CONSTRAINT test_table_array_pid PRIMARY KEY (test_table_array_id, test_table_array_updated));

SELECT sys_syn.in_table_add (
                'user_data',
                'test_table_array',
                'in',
                NULL,
                ARRAY[
                       $COL$("test_table_array_id","integer",ID,"in_source.test_table_array_id",,,,)$COL$,
                       $COL$("test_table_array_updated","timestamp with time zone",Attribute,"in_source.test_table_array_updated",1,,,)$COL$,
                       $COL$("test_table_array_text","text",Attribute,"in_source.test_table_array_text",,,,)$COL$
                ]::sys_syn.create_in_column[],
                'user_data.test_table_array',
                NULL
        );

INSERT INTO user_data.test_table_array(
        test_table_array_id, test_table_array_updated,             test_table_array_text)
VALUES  (1,              '2009-01-02 03:04:05-00',       'test_data1 v1'),
        (1,              '2010-01-02 03:04:05-00',       'test_data1 v2'),
        (2,              '2011-01-02 03:04:05-00',       'test_data2 v1'),
        (2,              '2012-01-02 03:04:05-00',       'test_data2 v2');

SELECT sys_syn.out_table_add('user_data', 'test_table_array', 'out', data_view => TRUE);

SELECT user_data.test_table_array_pull(FALSE);
SELECT user_data.test_table_array_out_move();

SELECT * FROM user_data.test_table_array_out_queue_data;
