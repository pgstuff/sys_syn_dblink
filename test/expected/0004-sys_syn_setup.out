CREATE EXTENSION sys_syn;
CREATE SCHEMA user_data
    AUTHORIZATION postgres;
CREATE TABLE user_data.test_table (
        test_table_id integer NOT NULL,
        test_table_text text,
        CONSTRAINT test_table_pid PRIMARY KEY (test_table_id));
INSERT INTO sys_syn.in_groups_def VALUES ('in');
DO $$BEGIN
        EXECUTE sys_syn.in_table_create_sql('user_data.test_table'::regclass, 'in');
END$$;
INSERT INTO user_data.test_table(
        test_table_id, test_table_text)
VALUES  (1,              'test_data 1'),
        (2,              'test_data 2'),
        (3,              'test_data 3');
INSERT INTO sys_syn.out_groups_def VALUES ('out');
INSERT INTO sys_syn.out_groups_def VALUES ('out2');
SELECT sys_syn.out_table_create('user_data', 'test_table', 'out', data_view => TRUE);
 out_table_create 
------------------
 
(1 row)

SELECT sys_syn.out_table_create('user_data', 'test_table', 'out2');
 out_table_create 
------------------
 
(1 row)

SELECT user_data.test_table_pull(FALSE);
 test_table_pull 
-----------------
 t
(1 row)

SELECT user_data.test_table_out_move();
 test_table_out_move 
---------------------
 t
(1 row)

SELECT id, delta_type, queue_state FROM user_data.test_table_out_queue;
 id  | delta_type | queue_state 
-----+------------+-------------
 (1) | Add        | Unclaimed
 (2) | Add        | Unclaimed
 (3) | Add        | Unclaimed
(3 rows)

CREATE EXTENSION hstore;
CREATE EXTENSION dblink;
CREATE TABLE public.test_data (
        test_data_id integer NOT NULL,
        test_data_text text,
        CONSTRAINT test_data_pid PRIMARY KEY (test_data_id));
INSERT INTO sys_syn.in_groups_def VALUES ('group');
DO $$BEGIN
        EXECUTE sys_syn.in_table_create_sql('public.test_data'::regclass, 'group');
END$$;
INSERT INTO public.test_data(
        test_data_id, test_data_text)
VALUES  (1,              'test_data 1'),
        (2,              'test_data 2'),
        (3,              'test_data 3');
INSERT INTO sys_syn.out_groups_def VALUES ('group');
SELECT sys_syn.out_table_create('public', 'test_data', 'group', data_view => TRUE);
 out_table_create 
------------------
 
(1 row)

SELECT public.test_data_pull(FALSE);
 test_data_pull 
----------------
 t
(1 row)

SELECT public.test_data_group_move();
 test_data_group_move 
----------------------
 t
(1 row)

SELECT id, delta_type, queue_state FROM public.test_data_group_queue;
 id  | delta_type | queue_state 
-----+------------+-------------
 (1) | Add        | Unclaimed
 (2) | Add        | Unclaimed
 (3) | Add        | Unclaimed
(3 rows)

CREATE SCHEMA "User Data"
    AUTHORIZATION postgres;
CREATE TABLE "User Data"."Test Table" (
        "Test Table Id" integer NOT NULL,
        "Test Table Text" text,
        CONSTRAINT "Test Table_pid" PRIMARY KEY ("Test Table Id"));
INSERT INTO sys_syn.in_groups_def VALUES ('In Group');
SELECT sys_syn.in_table_create_sql('"User Data"."Test Table"'::regclass, 'In Group');
                                            in_table_create_sql                                            
-----------------------------------------------------------------------------------------------------------
 SELECT sys_syn.in_table_create(                                                                          +
                 schema          => '"User Data"'::regnamespace,                                          +
                 in_table_id     => 'Test Table',                                                         +
                 in_group_id     => 'In Group',                                                           +
                 in_pull_id      => NULL,                                                                 +
                 in_columns      => ARRAY[                                                                +
                        $COL$("Test Table Id","integer",Id,"in_source.\"Test Table Id\"",,,,)$COL$,       +
                        $COL$("Test Table Text","text",Attribute,"in_source.\"Test Table Text\"",,,,)$COL$+
                 ]::sys_syn.create_in_column[],                                                           +
                 full_table_reference    => '"User Data"."Test Table"',                                   +
                 changes_table_reference => NULL,                                                         +
                 full_sql                => NULL,                                                         +
                 changes_sql             => NULL,                                                         +
                 full_pre_sql            => NULL,                                                         +
                 changes_pre_sql         => NULL,                                                         +
                 full_post_sql           => NULL,                                                         +
                 changes_post_sql        => NULL,                                                         +
                 enable_deletes_implied  => TRUE,                                                         +
                 full_prepull_id         => NULL,                                                         +
                 changes_prepull_id      => NULL                                                          +
         );
(1 row)

DO $$BEGIN
        EXECUTE sys_syn.in_table_create_sql('"User Data"."Test Table"'::regclass, 'In Group');
END$$;
INSERT INTO "User Data"."Test Table"(
        "Test Table Id", "Test Table Text")
VALUES (1,              'test_data v1');
INSERT INTO sys_syn.out_groups_def VALUES ('Out Group');
DO $$BEGIN
        EXECUTE sys_syn.out_table_create_sql('"User Data"'::regnamespace, 'Test Table', 'Out Group', data_view => TRUE);
END$$;
SELECT "User Data"."Test Table_pull"(FALSE);
 Test Table_pull 
-----------------
 t
(1 row)

SELECT "User Data"."Test Table_Out Group_move"();
 Test Table_Out Group_move 
---------------------------
 t
(1 row)

CREATE TABLE user_data.test_table_array (
        test_table_array_id integer NOT NULL,
        test_table_array_updated timestamp with time zone,
        test_table_array_text text,
        CONSTRAINT test_table_array_pid PRIMARY KEY (test_table_array_id, test_table_array_updated));
SELECT sys_syn.in_table_create (
                'user_data',
                'test_table_array',
                'in',
                NULL,
                ARRAY[
                       $COL$("test_table_array_id","integer",Id,"in_source.test_table_array_id",,,,)$COL$,
                       $COL$("test_table_array_updated","timestamp with time zone",Attribute,"in_source.test_table_array_updated",1,,,)$COL$,
                       $COL$("test_table_array_text","text",Attribute,"in_source.test_table_array_text",,,,)$COL$
                ]::sys_syn.create_in_column[],
                'user_data.test_table_array',
                NULL
        );
 in_table_create 
-----------------
 
(1 row)

INSERT INTO user_data.test_table_array(
        test_table_array_id, test_table_array_updated,             test_table_array_text)
VALUES  (1,              '2009-01-02 03:04:05-00',       'test_data1 v1'),
        (1,              '2010-01-02 03:04:05-00',       'test_data1 v2'),
        (2,              '2011-01-02 03:04:05-00',       'test_data2 v1'),
        (2,              '2012-01-02 03:04:05-00',       'test_data2 v2');
SELECT sys_syn.out_table_create('user_data', 'test_table_array', 'out', data_view => TRUE);
 out_table_create 
------------------
 
(1 row)

SELECT user_data.test_table_array_pull(FALSE);
 test_table_array_pull 
-----------------------
 t
(1 row)

SELECT user_data.test_table_array_out_move();
 test_table_array_out_move 
---------------------------
 t
(1 row)

SELECT * FROM user_data.test_table_array_out_queue_data;
 sys_syn_id | sys_syn_trans_id_in | sys_syn_delta_type | sys_syn_queue_state | sys_syn_queue_id | sys_syn_queue_priority | sys_syn_hold_updated | sys_syn_hold_trans_id_first | sys_syn_hold_trans_id_last | sys_syn_hold_reason_count | sys_syn_hold_reason_id | sys_syn_hold_reason_text | sys_syn_trans_id_out | sys_syn_processed_time | sys_syn_attribute_array_ordinal | test_table_array_id |   test_table_array_updated   | test_table_array_text 
------------+---------------------+--------------------+---------------------+------------------+------------------------+----------------------+-----------------------------+----------------------------+---------------------------+------------------------+--------------------------+----------------------+------------------------+---------------------------------+---------------------+------------------------------+-----------------------
 (1)        |                   7 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               1 |                   1 | Thu Jan 01 19:04:05 2009 PST | test_data1 v1
 (1)        |                   7 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               2 |                   1 | Fri Jan 01 19:04:05 2010 PST | test_data1 v2
 (2)        |                   7 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               1 |                   2 | Sat Jan 01 19:04:05 2011 PST | test_data2 v1
 (2)        |                   7 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               2 |                   2 | Sun Jan 01 19:04:05 2012 PST | test_data2 v2
(4 rows)

CREATE TABLE user_data.test_table_bitemporal (
        test_table_bitemporal_id integer NOT NULL,
        test_table_bitemporal_updated timestamp with time zone,
        test_table_bitemporal_start date,
        test_table_bitemporal_end date,
        test_table_bitemporal_text text,
        CONSTRAINT test_table_bitemporal_pid PRIMARY KEY (test_table_bitemporal_id, test_table_bitemporal_updated, test_table_bitemporal_start));
SELECT sys_syn.in_table_create (
                'user_data',
                'test_table_bitemporal',
                'in',
                NULL,
                ARRAY[
                       $COL$("test_table_bitemporal_id","integer",Id,"in_source.test_table_bitemporal_id",,,,)$COL$,
                       $COL$("test_table_bitemporal_updated","timestamp with time zone",Attribute,"in_source.test_table_bitemporal_updated",1,,,)$COL$,
                       $COL$("test_table_bitemporal_start","date",Attribute,"in_source.test_table_bitemporal_start",2,,,)$COL$,
                       $COL$("test_table_bitemporal_end","date",Attribute,"in_source.test_table_bitemporal_end",3,,,)$COL$,
                       $COL$("test_table_bitemporal_text","text",Attribute,"in_source.test_table_bitemporal_text",,,,)$COL$
                ]::sys_syn.create_in_column[],
                'user_data.test_table_bitemporal',
                NULL
        );
 in_table_create 
-----------------
 
(1 row)

INSERT INTO user_data.test_table_bitemporal(
        test_table_bitemporal_id, test_table_bitemporal_updated, test_table_bitemporal_start, test_table_bitemporal_end, test_table_bitemporal_text)
VALUES  (1,              '2009-01-02 03:04:05-00',       '2009-01-03',  '9999-12-31',   'test_data1 no changes'),
        (2,              '2011-01-02 03:04:05-00',       '2009-01-01',  '9999-12-31',   'test_data2 v1 original'),
        (2,              '2012-01-02 03:04:05-00',       '2009-02-01',  '9999-12-31',   'test_data2 v2 forward period_2 change'),
        (3,              '2011-01-02 03:04:05-00',       '2009-01-01',  '9999-12-31',   'test_data3 v1 original'),
        (3,              '2012-01-02 03:04:05-00',       '2009-01-01',  '9999-12-31',   'test_data3 v2 replace period_2 change'),
        (4,              '2011-01-02 03:04:05-00',       '2009-01-01',  '9999-12-31',   'test_data4 v1 original'),
        (4,              '2012-01-02 03:04:05-00',       '2008-12-01',  '9999-12-31',   'test_data4 v2 backdate period_2 change'),
        (5,              '2011-01-02 03:04:05-00',       '2009-01-01',  '9999-12-31',   'test_data5 v1 original'),
        (5,              '2012-01-02 03:04:05-00',       '2009-01-01',  '9999-12-31',   'test_data5 v2 replace period_2 change'),
        (5,              '2013-01-02 03:04:05-00',       '2009-03-01',  '9999-12-31',   'test_data5 v3 forward period_2 change'),
        (5,              '2014-01-02 03:04:05-00',       '2009-02-01',  '9999-12-31',   'test_data5 v4 backdate period_2 change');
SELECT sys_syn.out_table_create('user_data', 'test_table_bitemporal', 'out', data_view => TRUE);
 out_table_create 
------------------
 
(1 row)

SELECT user_data.test_table_bitemporal_pull(FALSE);
 test_table_bitemporal_pull 
----------------------------
 t
(1 row)

SELECT user_data.test_table_bitemporal_out_move();
 test_table_bitemporal_out_move 
--------------------------------
 t
(1 row)

SELECT * FROM user_data.test_table_bitemporal_out_queue_data;
 sys_syn_id | sys_syn_trans_id_in | sys_syn_delta_type | sys_syn_queue_state | sys_syn_queue_id | sys_syn_queue_priority | sys_syn_hold_updated | sys_syn_hold_trans_id_first | sys_syn_hold_trans_id_last | sys_syn_hold_reason_count | sys_syn_hold_reason_id | sys_syn_hold_reason_text | sys_syn_trans_id_out | sys_syn_processed_time | sys_syn_attribute_array_ordinal | test_table_bitemporal_id | test_table_bitemporal_updated | test_table_bitemporal_start | test_table_bitemporal_end |       test_table_bitemporal_text       
------------+---------------------+--------------------+---------------------+------------------+------------------------+----------------------+-----------------------------+----------------------------+---------------------------+------------------------+--------------------------+----------------------+------------------------+---------------------------------+--------------------------+-------------------------------+-----------------------------+---------------------------+----------------------------------------
 (1)        |                   9 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               1 |                        1 | Thu Jan 01 19:04:05 2009 PST  | 01-03-2009                  | 12-31-9999                | test_data1 no changes
 (2)        |                   9 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               1 |                        2 | Sat Jan 01 19:04:05 2011 PST  | 01-01-2009                  | 12-31-9999                | test_data2 v1 original
 (2)        |                   9 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               2 |                        2 | Sun Jan 01 19:04:05 2012 PST  | 02-01-2009                  | 12-31-9999                | test_data2 v2 forward period_2 change
 (3)        |                   9 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               1 |                        3 | Sat Jan 01 19:04:05 2011 PST  | 01-01-2009                  | 12-31-9999                | test_data3 v1 original
 (3)        |                   9 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               2 |                        3 | Sun Jan 01 19:04:05 2012 PST  | 01-01-2009                  | 12-31-9999                | test_data3 v2 replace period_2 change
 (4)        |                   9 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               1 |                        4 | Sat Jan 01 19:04:05 2011 PST  | 01-01-2009                  | 12-31-9999                | test_data4 v1 original
 (4)        |                   9 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               2 |                        4 | Sun Jan 01 19:04:05 2012 PST  | 12-01-2008                  | 12-31-9999                | test_data4 v2 backdate period_2 change
 (5)        |                   9 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               1 |                        5 | Sat Jan 01 19:04:05 2011 PST  | 01-01-2009                  | 12-31-9999                | test_data5 v1 original
 (5)        |                   9 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               2 |                        5 | Sun Jan 01 19:04:05 2012 PST  | 01-01-2009                  | 12-31-9999                | test_data5 v2 replace period_2 change
 (5)        |                   9 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               3 |                        5 | Tue Jan 01 19:04:05 2013 PST  | 03-01-2009                  | 12-31-9999                | test_data5 v3 forward period_2 change
 (5)        |                   9 | Add                | Unclaimed           |                  |                        |                      |                             |                            |                           |                        |                          |                      |                        |                               4 |                        5 | Wed Jan 01 19:04:05 2014 PST  | 02-01-2009                  | 12-31-9999                | test_data5 v4 backdate period_2 change
(11 rows)

